### PR TITLE
COMMERCE-8937 - Commerce should always use portalImpl.getPlidFromPortletId instead of their own method.

### PR DESCRIPTION
**Pull request by ethib137**

### CODE REVIEW

**Comment by ethib137** [comment]
cc: @ealonso

**Comment by ethib137** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
ci:test:sf

**Comment by ethib137** [comment]
ci:test:relevant

**Review by alessiorendina — CHANGES_REQUESTED** [review]
Hi @ethib137 it looks good to me, I left just a comment. Thank you!

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/helper/AssetDisplayPageFriendlyURLResolverHelper.java [hunk 1]
```diff
@@ -40,104 +24,8 @@
...
-		catch (Exception exception) {
-			if (_log.isWarnEnabled()) {
-				_log.warn(exception);
-			}
-		}
-
-		return LayoutConstants.DEFAULT_PLID;
-	}
-
 	public int getURLSeparatorLength(String urlSeparator) {
```
[the remark this answers is not in the document]
Comment (ethib137) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/helper/AssetDisplayPageFriendlyURLResolverHelper.java]: Yeah, I can do that. :)

**Comment by ethib137** [comment]
ci:test:sf

**Comment by ethib137** [comment]
ci:test:relevant

**Comment by ethib137** [comment]
@alessiorendina I've updated the pr according to your comments. Let me know if you see anything else that should be changed. :)

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/CPDefinitionAssetDisplayPageFriendlyURLResolver.java [hunk 2]
```diff
@@ -94,9 +93,7 @@ public String getActualURL(
 
 		long classNameId = _portal.getClassNameId(CProduct.class);
 
-		String urlTitle = friendlyURL.substring(
-			_assetDisplayPageFriendlyURLResolverHelper.getURLSeparatorLength(
-				getURLSeparator()));
+		String urlTitle = friendlyURL.substring(getURLSeparator().length());
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/CPDefinitionAssetDisplayPageFriendlyURLResolver.java:96]: Same here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@ethib137 Just a little SF thing. It is strange that the SF task didn't catch it. Try to run it, please.

**Comment by ethib137** [comment]
Hi @alessiorendina, happy to do that. Yeah I thought it was odd as well that SF didn't mind it. We do chaining like that a few other places in portal which you can see if you search for `().length()`. Still, probably better without. I'll push the change shortly.

**Comment by ethib137** [comment]
ci:test:sf

**Comment by ethib137** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/AssetCategoryAssetDisplayPageFriendlyURLResolver.java [hunk 3]
```diff
@@ -84,9 +83,7 @@ public String getActualURL(
 
 		Group companyGroup = _groupLocalService.getCompanyGroup(companyId);
 
-		String urlTitle = friendlyURL.substring(
-			_assetDisplayPageFriendlyURLResolverHelper.getURLSeparatorLength(
-				getURLSeparator()));
+		String urlTitle = friendlyURL.substring(getURLSeparator().length());
```
[the remark this answers is not in the document]
Comment (ethib137) [reply modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/asset/display/page/portlet/AssetCategoryAssetDisplayPageFriendlyURLResolver.java]: Done.

**Comment by ethib137** [comment]
@alessiorendina Updated with the change to remove chaining.

**Comment by ethib137** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Review by alessiorendina — APPROVED** [review]
It looks good to me, thank you @ethib137

**Comment by ethib137** [comment]
Hi @alessiorendina any update on this? Can it be forwarded?

**Comment by iovaneale** [comment]
ci:forward

**Pull request closed without merging**